plugins/woocommerce/src/StoreApi/Utilities/CartController.php

```diff
@@ -486,13 +486,29 @@ public function validate_cart() {
 		}
 	}
 
+	/**
+	 * When placing an order, validate that the cart is not empty.
+	 *
+	 * @throws InvalidCartException Exception if the cart is empty.
+	 */
+	public function validate_cart_not_empty() {
```

Comment:
I originally implemented the validation inside `validate_cart_items` but that function also gets called for the cart block (which makes sense). I've opted to separate out this validation as it's only relevant when you go to place an order.